### PR TITLE
TP-788: Fix edits after publishing workbasket

### PR DESCRIPTION
**Pull request by andyhd**
After successfully editing and publishing a change, subsequent edits do not appear in
the list of tariff changes on the home page.

This is due to the workbasket not being updated in the user session. Subsequent edits
are appended to the already-published workbasket. The home page only lists the contents
of the current workbasket, which is a new and empty one after publishing the last one.

This change ensures that the session is updated on publishing the workbasket, and that
subsequent edits are correctly appended to the new unpublished workbasket.

Before testing make sure to clear your session by logging out and back in again.

### CODE REVIEW

File: common/forms.py [hunk 1]
```diff
@@ -17,7 +17,10 @@ class DateInputFieldFixed(DateInputField):
     def compress(self, data_list):
         day, month, year = data_list or [None, None, None]
         if day and month and year:
-            return date(day=int(day), month=int(month), year=int(year))
+            try:
+                return date(day=int(day), month=int(month), year=int(year))
+            except ValueError as e:
+                raise ValidationError(str(e).capitalize()) from e
```
Comment (andyhd) [inline common/forms.py:23]: This is an unrelated fix for validating date inputs - previously, you could enter invalid day and month values (eg: day > 31 or month > 12) and you would get a 500 error instead of a form error.

File: workbaskets/views/__init__.py [hunk 2]
```diff
@@ -26,55 +25,41 @@ class WorkBasketViewSet(viewsets.ModelViewSet):
...
+    permission_required = "workbaskets.change_workbasket"
 
-    workbasket.export_to_cds()
-    workbasket.save()
+    @transaction.atomic
+    def get_redirect_url(self, *args, **kwargs):
+        workbasket: WorkBasket = self.get_object()
 
-    return redirect("index")
+        workbasket.submit_for_approval()
```
Comment (andyhd) [inline workbaskets/views/__init__.py:59]: `submit_for_approval` calls `workbasket.full_clean`

**Review by stuaxo — APPROVED** [review]
Nice view changes in this too.
